sdk/core/loader/boot.cc

```diff
@@ -568,14 +574,11 @@ namespace
                         }
                         return false;
 					};
-					bool found = false;
-					for (auto &compartment : image.privilegedCompartments)
+					bool found = findExport(image.allocator());
+					if (!found && findExport(image.scheduler()))
```

Comment:
It's possibly a bit of a pity that the scheduler isn't represented as a `PrivilegedCompartment` and so we have to special-case these two.

Reply:
They are both `PrivilegedCompartment`s, the old code just iterated over them, but since we're now doing different things for the two privileged compartments that doesn't make sense anymore.